### PR TITLE
Add download / unzip of code archive

### PR DESCRIPTION
**Pull request by dfangl**
## Motivation
Provide mechanism to download code archive via an url instead of it being mounted/copied into the container before startup. This is necessary for environments where mounting/copying is not feasible.

## Changes
* If the variable "CODE_ARCHIVE_DOWNLOAD_URL" is set, download the zip archive from that url and unzip it to `/var/task`.

### CODE REVIEW

File: cmd/localstack/awsutil.go [hunk 1]
```diff
@@ -128,6 +132,70 @@ func GetenvWithDefault(key string, defaultValue string) string {
...
+	}
+	defer r.Close()
+	for _, f := range r.File {
+		rc, err := f.Open()
+		if err != nil {
+			log.Fatal(err)
+		}
+		target_file_name := path.Join(taskFolder, f.Name)
+		if f.FileInfo().IsDir() {
+			err = os.MkdirAll(target_file_name, 0777)
```
Comment (dominikschubert) [inline cmd/localstack/awsutil.go]:
```suggestion
			err = os.MkdirAll(target_file_name, os.ModePerm)
```
for consistency

File: cmd/localstack/main.go [hunk 2]
```diff
@@ -33,6 +34,8 @@ func InitLsOpts() *LsOpts {
 		// optional with default
 		InteropPort:     GetenvWithDefault("LOCALSTACK_INTEROP_PORT", "9563"),
 		InitTracingPort: GetenvWithDefault("LOCALSTACK_RUNTIME_TRACING_PORT", "9564"),
+		// optional or empty
+		CodeDownloadUrl: os.Getenv("CODE_ARCHIVE_DOWNLOAD_URL"),
```
Comment (dominikschubert) [inline cmd/localstack/main.go]: I feel like we should still namespace this to avoid any (admittedly unlikely) conflicts.

**Review by dominikschubert — APPROVED** [review]
LGTM, minor nits included

File: cmd/localstack/awsutil.go (hunk 1 above)
Comment (dfangl) [reply cmd/localstack/awsutil.go]: Ah, overlooked that one, thanks!

File: cmd/localstack/main.go (hunk 2 above)
Comment (dfangl) [reply cmd/localstack/main.go]: Agree!

**Pull request merged**